app/workers/detect_invariants.rb

```diff
@@ -23,5 +24,23 @@ def detect_application_choices_in_old_states
     end
   end
 
+  def detect_outstanding_references_on_submitted_applications
+    applications_with_reference_weirdness = ApplicationChoice
+      .joins(application_form: [:application_references])
+      .where.not(application_choices: { status: 'unsubmitted' })
+      .where(references: { feedback_status: :feedback_requested })
```

Comment:
Now that candidates can add more than two `feedback_requested` references before submission is it possible for a submitted application to have two `feedback_provided` and one or more `feedback_requested` references?

(I couldn't see any logic for cancelling 'spare' pending references in the submit application logic but I might be missing it).

Reply:
It shouldn't be possible, following this change https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/app/services/submit_reference.rb#L11

Reply:
I thought that logic must be somewhere. Makes sense.